oidc/verify.go

```diff
@@ -250,12 +253,24 @@ func (v *IDTokenVerifier) Verify(ctx context.Context, rawIDToken string) (*IDTok
 
 	// Check issuer.
 	if !v.config.SkipIssuerCheck && t.Issuer != v.issuer {
-		// Google sometimes returns "accounts.google.com" as the issuer claim instead of
-		// the required "https://accounts.google.com". Detect this case and allow it only
-		// for Google.
-		//
-		// We will not add hooks to let other providers go off spec like this.
-		if !(v.issuer == issuerGoogleAccounts && t.Issuer == issuerGoogleAccountsNoScheme) {
+		if v.config.AdfsCompatibility {
+			// ADFS appends the string "/services/trust" to the issuer string, breaking the standard.
+			// More details can be found here:
+			// https://github.com/aspnet/Security/issues/1852
+			expectedIssuer := v.issuer + "/services/trust"
+			if t.Issuer != expectedIssuer {
```

Comment:
do we need to handle the case of the issuer being put in the `access_token_issuer` claim instead of the `issuer` claim for ADFS?

Reply:
As far as I understand the description of the issue here this shouldn't be necessary: https://github.com/aspnet/Security/issues/1852#issuecomment-426506324

however, i've just read the description again and I noticed now that depending on which ADFS endpoint was used to generate an access token that `/services/trust` string might or might not get appended. So I should change my implementation in such a way that if ADFS compatibility is enabled it just accepts both issuers, with and without the `/services/trust` suffix.

Reply:
specifically, it says that here: https://github.com/aspnet/Security/issues/1852#issuecomment-426506324

Reply:
added here: https://github.com/grafana/go-oidc/pull/1/commits/00cd75ed6582066783f028e2a54ece3c32b39e66